gamemodes/terrortown/gamemode/client/cl_equip.lua

```diff
@@ -832,12 +843,19 @@ function TraitorMenuPopup()
 		local weapon = choice.id
 		local steamid = ply:SteamID64()
 
-		CreateFavTable()
-
 		if pnl.favorite then
-			RemoveFavorite(steamid, role, weapon)
+			local favorite = equipFavModel:Find({steamid, role, weapon})
+			favorite:Delete()
+			-- possible one-liner alternative:
+			--equipFavModel:Find({steamid, role, weapon}):Delete()
```

Comment:
creates an object of the model with values from the database and then deletes it (maybe needs an `IsValid` check just to be safe)